06-walkthrough-1.Rmd

```diff
@@ -348,12 +654,59 @@ m_course <- lmer(percentage_earned ~ TimeSpent + (1|course_id), data = dat)
 summary(m_course)
 ```
 
-A common way to understand how much variability is at the group level is to calculate the *intra-class* correlation. This value is the proportion of the variability in the outcome (the *y*-variable) that is accounted for solely by the groups identified in the model. There is a useful function in the **sjstats** package for doing this.
+As we mentioned earlier, a common way to understand how much variability is at 
+the group level is to calculate the *intra-class* correlation. This value is the
+proportion of the variability in the outcome (the *y*-variable) that is 
+accounted for solely by the groups identified in the model. There is a useful 
+function in the **performance** package for doing this.
 
 ```{r}
-# install.packages("sjstats")
-library(sjstats)
+# install.packages("performance")
+library(performance)
 icc(m_course)
 ```
 
-This shows that nearly 17% of the variability in the percentage of points students earned can be explained simply by knowing what class they are in. 
+This shows that nearly 17% of the variability in the percentage of points 
+students earned can be explained simply by knowing what class they are in. 
+
+There is much more to do with multi-level models. We briefly discuss a 
+common extension to the model we just used - although such a model, in which
+the effects of being a student in a class are recognized through specifying 
+what the groups are, is very common. The extensions is for adding 
+additional levels.
+
+The data that we are using is all from one school, and so we cannot estimate a 
+"two-level" model. Imagine, however, that instead of 26 classes, we had data 
+from students from 230 classes, and that these classes were from 15 schools. We 
+could estimate a two-level, varying intercepts (where there are now two groups 
+with effects) model very similar to the model we estimated above, but simply 
+with another group added for the school. The model will account for the way in 
+which the classes are nested within the schools automatically (Bates, Maechler, 
+Bolker, & Walker, 2015). 
+
+```{r}
+m_course_school <- lmer(percentage_earned ~ TimeSpent + (1|course_id) + (1|school_id), data = dat)
+```
+
+Were we to estimate this model (and then use the `icc()` function), we would 
+see two ICC values representing the proportion of the variation in the dependent
+variable explained by each of the two groups we added - the course *and* the
+school. You can add further still levels to the model, as the {lme4} package was
+designed for complex multi-level models (and even those with not nested, but
+crossed random effects; a topic beyond the scope of this walkthrough, but which 
+is described in West, Welch, & Galecki, 2015).
+
+There is much more that can be done with multi-level models. For further 
```

Comment:
This sentence in 699 is very similar to the sentence in 672 - perhaps delete it in one of those places.